lib/Controller/ConfigController.php

```diff
@@ -66,28 +69,27 @@ public function setConfig(array $values): DataResponse {
 	 * @throws PreConditionNotMetException
 	 */
 	public function setSensitiveConfig(array $values): DataResponse {
+		$userConfig = UserConfig::fromArray($values);
+		$userConfig->toConfig($this->userId, $this->config);
+
 		// revoke the oauth token if needed
-		if (isset($values['token']) && $values['token'] === '') {
-			$tokenType = $this->config->getUserValue($this->userId, Application::APP_ID, 'token_type');
+		if ($userConfig->token === '') {
+			$tokenType = $this->config->getUserTokenType($this->userId);
 			if ($tokenType === 'oauth') {
 				$this->gitlabAPIService->revokeOauthToken($this->userId);
 			}
 		}
 
-		foreach ($values as $key => $value) {
-			$this->config->setUserValue($this->userId, Application::APP_ID, $key, $value);
-		}
-
 		$result = [];
 
-		if (isset($values['token'])) {
+		if ($userConfig->token !== null) {
```

Comment:
```php
if ($userConfig->token === '') {
```

null check also required here. And let's convert the line 85 (`if ($userConfig->token !== null) {`) into an `else` block

Reply:
sorry nvm